proposals/dictionary-expressions.md

```diff
@@ -165,47 +165,47 @@ From [Open Question 3](#open-question-3), we will support *dictionary elements*
 2. Use applicable instance indexer if available; otherwise report an error during construction (or conversion?).
 3. Use C#12 initialization always.
 
+## Dictionary types
+
+The following are considered *dictionary types*:
+
+* A *type* with an appropriate *[create method](#create-methods)* where:
+  * The *iteration type* is `KeyValuePair<TKey, TValue>` determined from a `GetEnumerator` instance method or enumerable interface.
+  * The *type* has an instance *indexer* where:
+    * The indexer has a single parameter with an identity conversion from the parameter type to `TKey`.
+    * There is an identity conversion from the indexer type to `TValue`.
```

Comment:
dumb question.  why we need an identity-conversion here.  As opposed to just say "a parameter type `TKey`"?

Reply:
importantly, if stating this as a conversion is substantively different, i'd love an example of where this would arise and how it would differ from the simpler version i stated.

Reply:
> why we need an identity-conversion here.  As opposed to just say "a parameter type `TKey`"?

The identity conversion requirement would cover differences with nullable reference types for instance.
```csharp
class MyDictionary<TKey, TValue> :
    IEnumerable<KeyValuePair<TKey, TValue>>
    where TKey : class
{
    public TValue this[TKey? key] { get; }
    // ...
}
```

Reply:
> The identity conversion requirement would allow differences with nullable reference types for instance.

Perfect.  Can a comment to that effect be added?

Reply:
Is it okay if the indexer return type is `ref TValue`?